airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_17_0.java

```diff
@@ -0,0 +1,130 @@
+ * or greater than 0.2.0
+ *
+ * This is because as part of Airbyte V0.17.0-alpha, we made changes to the JSON validation
+ * parsers/serializers/deserializers from all connectors to allow for unknown properties, thus
+ * bumping their versions to v0.2.0 as part of that release.
+ *
+ * This is required in case we want to perform modifications to the Airbyte protocol in the future
+ * by adding new properties. Connectors that are not using these additional properties yet, should
+ * ignore them but keep working as usual instead of throwing json validation errors.
+ */
+public class MigrationV0_17_0 extends BaseMigration implements Migration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MigrationV0_17_0.class);
+
+  private static final ResourceId STANDARD_SOURCE_DEFINITION_RESOURCE_ID =
+      ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_SOURCE_DEFINITION");
+  private static final ResourceId STANDARD_DESTINATION_DEFINITION_RESOURCE_ID =
+      ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_DESTINATION_DEFINITION");
+
+  private static final String DESTINATION_SEEDS = "migrations/migrationV0_17_0/airbyte_config/destination_definitions.yaml";
+  private static final String SOURCE_SEEDS = "migrations/migrationV0_17_0/airbyte_config/source_definitions.yaml";
+
+  private final Migration previousMigration;
+
+  public MigrationV0_17_0(Migration previousMigration) {
+    super(previousMigration);
+    this.previousMigration = previousMigration;
+  }
+
+  @Override
+  public String getVersion() {
+    return "0.17.0-alpha";
+  }
+
+  @Override
+  public Map<ResourceId, JsonNode> getOutputSchema() {
+    return previousMigration.getOutputSchema();
+  }
+
+  @Override
+  public void migrate(Map<ResourceId, Stream<JsonNode>> inputData, Map<ResourceId, Consumer<JsonNode>> outputData) {
+    try {
+      final JsonNode sourceConnectors = Yamls.deserialize(MoreResources.readResource(SOURCE_SEEDS));
+      final JsonNode destinationConnectors = Yamls.deserialize(MoreResources.readResource(DESTINATION_SEEDS));
+
+      for (final Map.Entry<ResourceId, Stream<JsonNode>> entry : inputData.entrySet()) {
+        final Consumer<JsonNode> recordConsumer = outputData.get(entry.getKey());
+
+        entry.getValue().forEach(r -> {
+          if (entry.getKey().equals(STANDARD_SOURCE_DEFINITION_RESOURCE_ID)) {
+            ((ObjectNode) r).set("dockerImageTag", getDockerImageTag(sourceConnectors, "sourceDefinitionId", r));
+          } else if (entry.getKey().equals(STANDARD_DESTINATION_DEFINITION_RESOURCE_ID)) {
+            ((ObjectNode) r).set("dockerImageTag", getDockerImageTag(destinationConnectors, "destinationDefinitionId", r));
+          }
+          recordConsumer.accept(r);
+        });
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private JsonNode getDockerImageTag(final JsonNode connectors, final String definitionId, final JsonNode node) {
```

Comment:
can you just make a map of connectorDefinitionId => connector in the migrate method so that you don't need to iterator through the connectors everytime?